src/polyfills/keyboardevent-key.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Original shim code: https://github.com/shvaikalesh/shim-keyboard-event-key
+
+import {hasOwn} from '../utils/object';
+
+const keys = {
+  Win: 'Meta',
+  Scroll: 'ScrollLock',
+  Spacebar: ' ',
+
+  Down: 'ArrowDown',
+  Left: 'ArrowLeft',
+  Right: 'ArrowRight',
+  Up: 'ArrowUp',
+
+  Del: 'Delete',
+  Apps: 'ContextMenu',
+  Esc: 'Escape',
+
+  Multiply: '*',
+  Add: '+',
+  Subtract: '-',
+  Decimal: '.',
+  Divide: '/',
+};
+
+/**
+ * Sets the KeyboardEvent#key shim if it does not exist.
```

Comment:
reading this code, I see
1- it expects `key` to exist on the proto chain but "fixes" it, is that correct? If so, comment does not match behaviour
2- any case where `key` may actually not exist that we need to handle?
3- this pollyfills all browsers. do we want to restrict to IE only?

Reply:
I think it is worth checking IE by duplicating the `/Trident|MSIE|IEMobile/i`.test(win.navigator.userAgent); check

Reply:
Done, thanks for the suggestion